In [1]:
#!pip install docplex
from docplex.mp.model import Model

In [2]:
m=Model(name='Phone Production')

# **Variables**

In [3]:
#Phone Production Variables
foldyphoneproduction=m.continuous_var(name='foldy phone production')
tinyphoneproduction=m.continuous_var(name='tiny phone production')

In [4]:
# Production Time Constants
foldyphonetime=1.5
tinyphonetime=2

In [5]:
foldyphonesaleprice=900
tinyphonesaleprice=1100

# **Constrains**

In [8]:
foldyphoneconstraint=m.add_constraint(foldyphoneproduction>=500)
tinyphoneconstraint=m.add_constraint(tinyphoneproduction>=200)

In [9]:
# Production Constraint
totalprodconstraint=m.add_constraint(m.sum([tinyphoneproduction*tinyphonetime + foldyphoneproduction*foldyphonetime])<=2999.5)

# **GOALS/OBJ**

In [10]:
m.maximize(foldyphoneproduction*foldyphonesaleprice + tinyphoneproduction*tinyphonesaleprice)

In [11]:
#!pip install cplex
solution=m.solve()

In [12]:
solution.display()

solution for: Phone Production
objective: 1779700.000
foldy phone production = 1733.000
tiny phone production = 200.000


# In this particular production optimization problem, we  acquired a production plan that has;
   # * $ 1779700 maximum  profit also; 
   # * 1733 foldy phones and 200 tiny phones in total